### PR TITLE
Explore: adds table result dataframes sorting to get response columns order based on queries order

### LINKED ISSUE #24731: Explore table panel gets Value #A and Value #B the wrong way roung
When running with two queries in explore, I'd expect Value #A to be the first and Value #B to be the second:

image

Version Grafana v7.1.0-91219227pre (bd42407)

### PR DESCRIPTION
**Pull request by Estrax**
**What this PR does / why we need it**:

Adds sorting of dataframes in Explore table result to make sure the rendered table has a correct order of columns - relevant to the order of queries.

**Which issue(s) this PR fixes**:

Fixes #24731

### CODE REVIEW

**Comment by Estrax** [comment]
Not sure how to write a proper test for that, as this fix covers a kinda tricky case - when we run multiple queries [queryA, queryB] and the first result that comes back is from the queryB, then anything from queryA. It changes the order of columns with values in the result table.

**Review by ivanahuckova — APPROVED** [review]
LGTM! 🌈

**Comment by aocenas** [comment]
@Estrax 
> Not sure how to write a proper test for that, as this fix covers a kinda tricky case - when we run multiple queries [queryA, queryB] and the first result that comes back is from the queryB, then anything from queryA. It changes the order of columns with values in the result table.

Not sure what specifically you mean. We already have some tests for the result processor so I assume it's possible to just initialize it with [queryB, queryA] and then see if that was ordered in the same way as if that would be [queryA, queryB]

**Comment by Estrax** [comment]
@aocenas Sure, I'll go that way and will open another PR with a test coverage of that soon.

File: public/app/features/explore/utils/ResultProcessor.ts [hunk 1]
```diff
@@ -50,7 +50,20 @@ export class ResultProcessor {
...
     }
 
-    const onlyTables = this.dataFrames.filter(frame => shouldShowInVisualisationType(frame, 'table'));
+    const onlyTables = this.dataFrames
+      .filter((frame: DataFrame) => shouldShowInVisualisationType(frame, 'table'))
+      .sort((frameA: DataFrame, frameB: DataFrame) => {
+        const frameARefId = frameA.refId!;
+        const frameBRefId = frameB.refId!;
+
+        if (frameARefId > frameBRefId) {
```
Comment (aocenas) [inline public/app/features/explore/utils/ResultProcessor.ts:59]: How do we create the refs? As far as I am aware they are just string right? Are they always created in a way that this sort hold, ie don't we create them at random?

Comment (Estrax) [reply public/app/features/explore/utils/ResultProcessor.ts:59]: Yes, they are strings. They come from the query field and are in order (alphabetical order according to the order of queries in Explore - as same as with queries in dashboards).

Comment (aocenas) [reply public/app/features/explore/utils/ResultProcessor.ts:59]: Yes but in this function https://github.com/grafana/grafana/blob/ae09ccbf7977c0c5e5b9f8a2b431582816eca82a/public/app/core/utils/query.ts#L4 it's not starting from the latest value it always start from A. So if you add two queries you have AB, then delete the first one you have B, add a new row you have BA and this sort will sort in the other way around from what is on the screen. Wouldn't it make sense to sort it based on what the ordering is in the reducer and not count on the refIDs being sequential?

**Pull request merged**